### DISCUSSION TITLE
Does gpg-tui replace pinenty program?

### DISCUSSION THREAD

**Discussion by sim4life** [orhun/gpg-tui#245] [Q&A]
Using Win11 WSL2 ubuntu 22.04, I got gpg, gnupg, gpg-agent and dependent programs installed. I used Nix home-manager v24.05 to install gpg-tui and enabled Nixpkgs version of gpg v2.4.5. There was a conflict as two `gpg-agent` were running at a time. So I removed the WSL2 apt version of gpg with --purge, which also removed pinentry program.
Do I need to install pinentry program separately from Nix home-manager OR does `gpg-tui` replace the functionality of pinentry program and I need to just change config.toml file?

**Comment by orhun** [comment]
Hello! `gpg-tui` does not replace the functionality of a pinentry program so you need to install one separately. `gpg-tui` is just a TUI on top of your GnuPG keyring for managing the keys :)
